packages/kbn-openapi-generator/src/template_service/template_service.ts

```diff
@@ -17,7 +17,12 @@ export const AVAILABLE_TEMPLATES = ['zod_operation_schema'] as const;
 export type TemplateName = (typeof AVAILABLE_TEMPLATES)[number];
 
 export interface ITemplateService {
-  compileTemplate: (templateName: TemplateName, context: GenerationContext) => string;
+  compileTemplate: (
+    templateName: TemplateName,
+    context: GenerationContext & {
+      sources: Array<{ sourcePath: string; generationContext: GenerationContext }>;
```

Comment:
My intent here was to avoid making `GenerationContext` into a recursive type, but yeah it was too confusing to have the context + generationContext properties and type mixing. I replaced this bit by separating the bundle generation context from the single file generation context and also creating a `compileBundleTemplate` function separate from `compileTemplate` - lmk if it looks better to you now. We had the `imports` field in the generation context for the bundles, but it didn't actually contain any data so it wasn't usable.

Reply:
My idea regarding `imports` is that it's quite simple. Context could have necessary import entries when constructed. Looking at `zod_operation_schema.handlebars` using `imports` we can see that this approach will add imports for any import entry presented in `imports`.

```
{{#each imports}}
import {
  {{#each this}}{{.}},{{/each}}
} from "{{@key}}"
{{/each}}
```


There is no obstacle to add there necessary entries like `${operationId}`RequestQueryInput` and 
 `${operationId}}RequestParamsInput`. Of course it'll require reworking generation logic a big to support multiple contexts.

I'd investigate this option but this is just a simplification idea and isn't a blocker for now.

Reply:
I think `compileBundleTemplate` is much clearer 👍